plenum/test/checkpoints/test_checkpoint_bounds_after_catchup.py

```diff
@@ -11,32 +14,28 @@ def test_upper_bound_of_checkpoint_after_catchup_is_divisible_by_chk_freq(
         chkFreqPatched, looper, txnPoolNodeSet,
         sdk_pool_handle, sdk_wallet_steward, sdk_wallet_client, tdir,
         tconf, allPluginsPath):
-    sdk_send_random_and_check(looper, txnPoolNodeSet, sdk_pool_handle,
-                              sdk_wallet_client, 4)
-
-    _, new_node = sdk_add_new_steward_and_node(
-        looper, sdk_pool_handle, sdk_wallet_steward,
-        'EpsilonSteward', 'Epsilon', tdir, tconf,
-        allPluginsPath=allPluginsPath)
-    txnPoolNodeSet.append(new_node)
-    looper.run(checkNodesConnected(txnPoolNodeSet))
-    waitNodeDataEquality(looper, new_node, *txnPoolNodeSet[:-1],
-                         exclude_from_check=['check_last_ordered_3pc_backup'])
+    lagging_node = txnPoolNodeSet[-1]
+    with delay_rules_without_processing(lagging_node.nodeIbStasher, cDelay()):
+        sdk_send_random_and_check(looper, txnPoolNodeSet, sdk_pool_handle,
+                                  sdk_wallet_client, tconf.Max3PCBatchSize * CHK_FREQ * 2 + 1)
+    ensure_all_nodes_have_same_data(looper, txnPoolNodeSet)
+    # waitNodeDataEquality(looper, lagging_node, *txnPoolNodeSet[:-1],
+    #                      exclude_from_check=['check_last_ordered_3pc_backup'])
     # Epsilon did not participate in ordering of the batch with EpsilonSteward
     # NYM transaction and the batch with Epsilon NODE transaction.
     # Epsilon got these transactions via catch-up.
 
     sdk_send_random_and_check(looper, txnPoolNodeSet, sdk_pool_handle,
-                              sdk_wallet_client, 1)
+                              sdk_wallet_client, 4 * tconf.Max3PCBatchSize)
 
     for replica in txnPoolNodeSet[0].replicas.values():
-        check_stable_checkpoint(replica, 5)
+        check_stable_checkpoint(replica, CHK_FREQ * 3)
 
     # TODO: This is failing now because checkpoints are not created after catchup.
     #  PBFT paper describes catch-up with per-checkpoint granularity, but otherwise
     #  quite similar to plenum implementation. Authors of PBFT state that after
     #  catching up nodes set low watermark to last caught up checkpoint, which is
     #  actually equivalent to declaring that checkpoint stable. This means that
     #  most probably we need this functionality in plenum.
-    # for replica in new_node.replicas.values():
+    # for replica in lagging_node.replicas.values():
     #    check_stable_checkpoint(replica, 5)
```

Comment:
I suspect this should eventually work. No immediate action needed, just taking note so that it doesn't get lost again...

Reply:
This the test with name "test_upper_bound_of_checkpoint_after_catchup_is_divisible_by_chk_freq" If we add a new node, it will be "checkpoint_after_view_change"